File: ruby/max_profit.rb
def max_profit(prices)
  max_profit = 0
  initial_index = 0

  prices.each do |price|
    if prices[initial_index] < price
      profit = price - prices[initial_index]
      max_profit = [max_profit, profit].max
    else
      initial_index = prices.index(price)
    end
  end

  max_profit
end

File: python/maxProfit.py
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        current_day_price = prices[0]
        profit = 0

        for next_day_price in prices[1:]:
            if (current_day_price > next_day_price):
                current_day_price = next_day_price
                continue
            
            profit = max((next_day_price - current_day_price, profit))
        
        return profit

